### PR TITLE
Redundant and misleading materilization part removed.

### PR DESCRIPTION
**Pull request by buraklayer**

### CODE REVIEW

**Review by mwitiderrick — COMMENTED** [review]
We might need to do this for all examples in this repo. 

Also, since the primary key is already defined in the data folder under dataset.yml, why do we need to define it in `https://github.com/layerml/examples/blob/main/titanic/data/passenger_features/dataset.yml` again

File: titanic/data/passenger_features/dataset.yml [hunk 1]
```diff
@@ -39,7 +39,3 @@ schema:
   # together.
   primary_keys: ["PassengerId"]
 
-materializations:
-  - type: table
-    target: layer
```
Comment (kocabiyikalper) [inline titanic/data/passenger_features/dataset.yml:44]: `target: layer` part is needed in order to tell the target datasource. But `type` and `table_name` are redundant

Comment (buraklayer) [reply titanic/data/passenger_features/dataset.yml:44]: Well, it is working now without having it. Why it is the case then?

**Comment by buraklayer** [comment]
> We might need to do this for all examples in this repo.
> 
> Also, since the primary key is already defined in the data folder under dataset.yml, why do we need to define it in `https://github.com/layerml/examples/blob/main/titanic/data/passenger_features/dataset.yml` again

It's a good point. Actually, I was thinking that it checks if each of .sql files contains PassangerId or not but turns out when we remove it completely or change it to something random, Layer does not complain about it.

File: titanic/data/passenger_features/dataset.yml (hunk 1 above)
Comment (kocabiyikalper) [reply titanic/data/passenger_features/dataset.yml:44]: If there is no target entered, we use `layer` as default for featuresets. But when user wants to store the data in a different place, it should be added here

**Pull request closed without merging**